### PR TITLE
Adds Flask-Security password and role handling.

### PR DESCRIPTION
**Pull request by oceanzus**
@FBRTMaka @DanielJMaher Please review and test.

These modifications allow for the management of the users via Flask-Admin and Flask-Security in a separate Flask app that will be rolled out early next week under a new repo.  We should perform a design review of the role/scope system in light of the new capability this provides and see if it will make the permission system a bit more intuitive to the OOI team.

As it is, these modifications stand on their own without the 'admin' app.

### CODE REVIEW

File: ooiservices/app/models.py [hunk 1]
```diff
@@ -1179,10 +1199,10 @@ def from_json(json):
         new_user = User()
         new_user.validate_email(email)
         new_user.validate_password(password, password2)
-        pass_hash = generate_password_hash(password)
```
Comment (FBRTMaka) [inline ooiservices/app/models.py]: Any reason this is being changed?

File: ooiservices/app/models.py [hunk 2]
```diff
@@ -1229,17 +1253,26 @@ def insert_user(username='admin',
...
+    # TODO: Revisit security concerns regarding the return of password
+    @hybrid_property
     def password(self):
-        raise AttributeError('password is not a readable attribute')
+        return self._password
 
-    #Store the hashed password.
     @password.setter
-    def password(self, password):
-        self.pass_hash = generate_password_hash(password)
```
Comment (FBRTMaka) [inline ooiservices/app/models.py]: This looks like it does the same thing

File: ooiservices/app/models.py [hunk 3]
```diff
@@ -1229,17 +1253,26 @@ def insert_user(username='admin',
...
+    # def password(self):
+    #     raise AttributeError('password is not a readable attribute')
+    #
+    # #Store the hashed password.
+    # @password.setter
+    # def password(self, password):
+    #     self.password = encrypt_password(password)
 
     def verify_password(self, password):
-        return check_password_hash(self.pass_hash, password)
```
Comment (FBRTMaka) [inline ooiservices/app/models.py]: again, same thing?

File: ooiservices/tests/test_user.py [hunk 4]
```diff
@@ -182,10 +182,11 @@ def test_user_insert(self):
         user_name = User.query.filter_by(user_name='admin').first()
         self.assertTrue(user_name.user_name == 'admin')
 
-    def test_password_tampering(self):
```
Comment (FBRTMaka) [inline ooiservices/tests/test_user.py]: as a security item, this needs to be a high priority 'todo'

**Comment by FBRTMaka** [comment]
What was the objective here? Not trying to be difficult.  I just want to get an idea of what sort of security improvements this provides.  I'm sure I'm not seeing the big plan yet.

**Comment by FBRTMaka** [comment]
@DanielJMaher @oceanzus , Could we meet about this 'separate flask app' to handle security.  I'm interested in the pros and cons and am excited to see how it'll help overall.  Sound interesting!

**Comment by DanielJMaher** [comment]
@FBRTMaka this has your blessing as well, right?

**Comment by DanielJMaher** [comment]
@FBRTMaka Can we merge?

**Comment by FBRTMaka** [comment]
:+1:

**Pull request merged**